libs/core/src/config.ts

```diff
@@ -0,0 +1,38 @@
+import * as dotenv from 'dotenv';
+import { ZodType, z } from 'zod';
+import { Environments } from './ports';
+
+dotenv.config({ path: '../../.env' });
+
+/**
+ * Extends target config with payload after validating schema
+ *
+ * @param target target payload
+ * @param extend extended payload
+ * @param schema extended schema
+ * @returns extended config
+ */
+export const configure = <T, E extends Record<string, unknown>>(
+  target: Readonly<T>,
+  extend: Readonly<E>,
+  schema: ZodType<E>,
+): Readonly<T & E> =>
+  Object.assign(target || {}, schema.parse(extend)) as Readonly<T & E>;
+
+const { NODE_ENV, PORT } = process.env;
+const port = parseInt(PORT || '8080', 10);
```

Comment:
I'm wondering if we should have a default value here given that we have different services that use the same `PORT` variable e.g. snapshot + discobot also have ports.